modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/CompactionListener.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.metastorage.server;
+
+/** Listener of the metastorage compaction. */
+@FunctionalInterface
+public interface CompactionListener {
+    /** Callback on completion of metastore compaction locally. */
+    void onCompleteLocally(long compactionRevision);
```

Comment:
I think that `onCompactionComplete` is a better name

Reply:
I improved your version a little.

Reply:
The new version is OK, but `onLocalCompactionComplete` is even better. But that's up to you.

Reply:
I'll leave it as is.